README.md
# FluidFox - Real-time log monitoring in your browser based in [NarrativeScience/Log.io](https://github.com/NarrativeScience/Log.io)

> WARNING: FluidFox its not yet production ready, use it with caution.

## Installation

`npm i -g fluidfox`

And you are ready, execute `fluidfox` to see the help page.

## Running the server

1. Create or edit a json file where your server configuration is going to live.

`> cat server.json`

```json
{
    "port": 28777
}
```

2. Run the server.

`fluidfox server server.json`

## Runing the harvester

1. Create or edit a json file where your harvester configuration is going to live.

`> cat harvester.json`

```json
{
    "nodeName": "application_server",
    "logStreams": {
        "application1": [
            {
                "name": "Custom name",
                "file": "/var/logs/obscure_file_name.log"
            }
        ],
        "nginx": [
            "/var/log/nginx/access.log",
            "/var/log/nginx/error.log"
        ]
    },
    "server": {
        "host": "your_server_here",
        "port": 28777
    }
}
```

2. Run the harvester.

`fluidfox harvester harvester.json`

Harvester will connect to the server and watch your files, when a file is modified, the new lines will be sended to the server.

## Running the web server

1. Create or edit a json file where your client configuration is going to live.

`> cat web.json`

```json
{
    "port": 8080,
    "server": {
        "ip": "your_server_here",
        "port": 28777
    }
}
```

2. Run the web server.

`fluidfox web web.json`

3. Browse to the IP/domain of the web server and access with the configured port, example http://127.0.0.1/8080

CHANGELOG
# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [1.0.2] - 2018-11-08
### Changed
- Send server connection config from web server to client.

## [1.0.1] - 2018-11-08
### Changed
- Version bump so npm its happy and lets me publish again the package.

## [1.0.0] - 2018-11-06
### Changed
- Refactored almost everything in typescript from the original Log.io.

### Removed
- All old files.

src/confSchema.ts
export const HarvesterJsonSchema = {
    definitions: {},
    $schema: "http://json-schema.org/draft-07/schema#",
    $id: "http://example.com/root.json",
    type: "object",
    title: "Harvester configuration schema",
    required: [
        "nodeName",
        "logStreams",
        "server",
    ],
    properties: {
        nodeName: {
            $id: "#/properties/nodeName",
            type: "string",
            title: "Nodename",
            default: "",
            examples: [
                "web_server",
            ],
            pattern: "^(.*)$",
        },
        logStreams: {
            $id: "#/properties/logStreams",
            type: "object",
            title: "Logger groups",
            minProperties: 1,
            patternProperties: {
                ".*": {
                    type: "array",
                    minItems: 1,
                    uniqueItems: true,
                    items: {
                        anyOf: [
                            {
                                type: "string",
                                examples: [
                                    "/var/log/nginx/access.log",
                                    "/var/log/nginx/error.log",
                                ]
                            },
                            {
                                type: "object",
                                required: [
                                    "name",
                                    "file"
                                ],
                                properties: {
                                    name: {
                                        type: "string"
                                    },
                                    file: {
                                        type: "string"
                                    }
                                },
                                examples: [
                                    {
                                        name: "Obscure log",
                                        file: "/var/log/fjdshf.log"
                                    }
                                ],
                            }
                        ]
                    },
                },
            },
        },
        server: {
            $id: "#/properties/server",
            type: "object",
            title: "Server connection config",
            required: [
                "host",
                "port",
            ],
            properties: {
                host: {
                    $id: "#/properties/server/properties/host",
                    type: "string",
                    title: "IP or hostname of the server",
                    default: "",
                    examples: [
                        "127.0.0.1",
                    ],
                    pattern: "^(.*)$",
                },
                port: {
                    $id: "#/properties/server/properties/port",
                    type: "integer",
                    title: "Port of the server",
                    default: 28777,
                    examples: [
                        28777,
                    ],
                },
            },
        },
    },
};

export const WebJsonSchema = {
    definitions: {},
    $schema: "http://json-schema.org/draft-07/schema#",
    $id: "http://example.com/root.json",
    type: "object",
    title: "Web server configuration schema",
    required: [
        "port",
        "server",
    ],
    properties: {
        port: {
            $id: "#/properties/port",
            type: "number",
            title: "Port of the web server",
            default: "",
            examples: [
                8080,
            ],
            pattern: "^[0-9]*$",
        },
        server: {
            $id: "#/properties/server",
            type: "object",
            title: "Server connection config",
            required: [
                "ip",
                "port",
            ],
            properties: {
                ip: {
                    $id: "#/properties/server/properties/ip",
                    type: "string",
                    title: "IP or hostname of the server",
                    default: "",
                    examples: [
                        "localhost",
                    ],
                    pattern: "^(.*)$",
                },
                port: {
                    $id: "#/properties/server/properties/port",
                    type: "integer",
                    title: "Port of the server",
                    default: 28777,
                    examples: [
                        28777,
                    ],
                },
            },
        },
    },
};
